test/controller.js

```diff
@@ -0,0 +1,371 @@
+
+    return {
+      memberToken,
+      ethersSafe,
+      podEnsRegistrar,
+    };
+  };
+
+  describe("when a pod has an admin", () => {
+    let { ethersSafe } = {};
+    beforeEach(async () => {
+      ({ ethersSafe } = await setup({ hasAdmin: true }));
+    });
+    it("should let admin set new admin", async () => {
+      await controller.connect(admin).updatePodAdmin(POD_ID, alice.address);
+      expect(await controller.podAdmin(POD_ID)).to.equal(alice.address);
+      const safe = await controller.podIdToSafe(POD_ID);
+      expect(await controller.areModulesLocked(safe)).to.equal(true);
+    });
+    it("should let admin remove admin", async () => {
+      await controller.connect(admin).updatePodAdmin(POD_ID, AddressZero);
+      expect(await controller.podAdmin(POD_ID)).to.equal(AddressZero);
+      const safe = await controller.podIdToSafe(POD_ID);
+      expect(await controller.areModulesLocked(safe)).to.equal(false);
+    });
+    it("should let admin unlock modules", async () => {
+      await controller.connect(admin).setPodModuleLock(POD_ID, false);
+
+      const safe = await controller.podIdToSafe(POD_ID);
+      expect(await controller.areModulesLocked(safe)).to.equal(false);
+    });
+    it("should throw if safe updates admin", async () => {
+      await expect(
+        safeExecutionHelper(ethersSafe, await controller.populateTransaction.updatePodAdmin(POD_ID, alice.address)),
+      ).to.be.revertedWith("GS013"); // sdk throws safe failure error
+    });
+    it("should throw if user updates module lock", async () => {
+      await expect(controller.connect(bob).setPodModuleLock(POD_ID, false)).to.be.revertedWith(
+        "Must be admin to set module lock",
+      );
+    });
+  });
+
+  describe("when a pod has no admin", () => {
+    it("should throw if member updates admin", async () => {
+      await setup();
+
+      await expect(controller.connect(alice).updatePodAdmin(POD_ID, alice.address)).to.be.revertedWith(
+        "Only safe can add new admin",
+      );
+    });
+
+    it("should let safe update admin", async () => {
+      const { ethersSafe } = await setup();
+      await safeExecutionHelper(ethersSafe, await controller.populateTransaction.updatePodAdmin(POD_ID, alice.address));
+
+      expect(await controller.podAdmin(POD_ID)).to.equal(alice.address);
+      const safe = await controller.podIdToSafe(POD_ID);
+      expect(await controller.areModulesLocked(safe)).to.equal(true);
+    });
+  });
+
+  describe("ejecting a safe", () => {
+    async function checkEject({ ethersSafe, memberToken, podId }) {
+      // Safe owners should be untouched.
+      expect(await ethersSafe.getOwners()).to.deep.equal([alice.address, bob.address]);
+      expect(await memberToken.balanceOf(alice.address, podId)).to.equal(0);
+      expect(await memberToken.balanceOf(bob.address, podId)).to.equal(0);
+
+      // // Checking if reverse resolver is zeroed.
```

Comment:
what's with all the commented code? not sure if it was like that before since we lost the diff

Reply:
its the ens tests, they will be refactored in another ticket